census_view.cpp

```diff
@@ -932,6 +936,844 @@ inline std::vector<std::string> const& CensusViewDataViewModel::all_headers() co
+    {
+        return typeid(table_sequence_converter).name();
+    }
+
+    virtual wxGridCellRenderer* create_renderer() const
+    {
+        return new(wx) wxGridCellStringRenderer();
+    }
+
+    virtual wxGridCellEditor* create_editor() const
+    {
+        return new(wx) DatumSequenceEditor();
+    }
+};
+
+// class table_int_range_converter
+
+class table_int_range_converter : public table_type_converter
+{
+  public:
+    char const* type() const override
+    {
+        return typeid(table_int_range_converter).name();
+    }
+
+    wxGridCellRenderer* create_renderer() const override
+    {
+        return new(wx) wxGridCellNumberRenderer();
+    }
+
+    wxGridCellEditor* create_editor() const override
+    {
+        return new(wx) RangeIntEditor();
+    }
+};
+
+// class table_double_range_converter
+
+class table_double_range_converter : public table_type_converter
+{
+  public:
+    char const* type() const override
+    {
+        return typeid(table_double_range_converter).name();
+    }
+
+    wxGridCellRenderer* create_renderer() const override
+    {
+        // Use wxGridCellStringRenderer instead of wxGridCellFloatRenderer to
+        // keep the number format as is.
+        return new(wx) wxGridCellStringRenderer();
+    }
+
+    wxGridCellEditor* create_editor() const override
+    {
+        return new(wx) RangeDoubleEditor();
+    }
+};
+
+// class table_date_converter
+
+class table_date_converter : public table_type_converter
+{
+  public:
+    wxString to_renderer_value(std::string const& value) const override
+    {
+        auto const date = ConvertDateToWx(value_cast<calendar_date>(value));
+        return date.FormatDate();
+    }
+
+    std::string from_editor_value(wxString const& value) const override
+    {
+        wxDateTime date;
+        auto date_parse_ok = date.ParseISODate(value);
+        LMI_ASSERT(date_parse_ok);
+
+        return value_cast<std::string>(ConvertDateFromWx(date));
+    }
+
+    char const* type() const override
+    {
+        return typeid(table_date_converter).name();
+    }
+
+    wxGridCellRenderer* create_renderer() const override
+    {
+        return new(wx) wxGridCellDateRenderer();
+    }
+
+    wxGridCellEditor* create_editor() const override
+    {
+        return new(wx) RangeDateEditor();
+    }
+};
+
+// class table_string_converter
+
+class table_string_converter : public table_type_converter
+{
+  public:
+    char const* type() const override
+    {
+        return typeid(table_string_converter).name();
+    }
+
+    wxString grid_value_type(any_member<Input> const&) const override
+    {
+        return wxGRID_VALUE_STRING;
+    }
+
+    void register_data_type(wxGrid*) const
+    {
+        // Do not register the data type because we use standard value type.
+    }
+};
+
+std::map<std::string, table_type_converter const*>&
+table_type_converter::get_all()
+{
+    static std::map<std::string, table_type_converter const*> all{
```

Comment:
This `{` should be on its own line.

Also, all commas should be first characters of the line.

I.e. I think the correct formatting would be:
```
static std::map<std::string, table_type_converter const*> all
    {
        { get_impl<xxx>().type()
        ,&get_impl<xxx>()
        }
    ,
        { get_impl<yyy>().type()
        ,&get_impl<yyy>()
        }
    }
```
although I'm not really sure if the inner items should be indented or not.